src/app/treelist-checkbox/treelist-checkbox.component.ts

```diff
@@ -0,0 +1,250 @@
+import { AfterViewInit, Component, EventEmitter, Input, OnInit, Output, ViewChild } from '@angular/core';
+import { DatasetNode } from 'app/dataset-node/dataset-node';
+@Component({
+  selector: 'gpf-treelist-checkbox',
+  templateUrl: './treelist-checkbox.component.html',
+  styleUrls: ['./treelist-checkbox.component.css']
+})
+export class TreelistCheckboxComponent implements OnInit, AfterViewInit {
```

Comment:
Perhaps this component should be renamed to something like `StudyFiltersTreeComponent`, since it is tightly coupled to the study filters logic at this moment.